test/index.test.js

```diff
@@ -156,4 +156,52 @@ describe('index', () => {
                     `${templateConfig.name}@${templateConfig.version} was successfully published`));
         });
     });
+
+    describe('Template Tag', () => {
+        const config = {
+            name: 'template/test',
+            tag: 'stable',
+            version: '1.0.0'
+        };
+
+        it('throws error when request yields an error', () => {
+            requestMock.rejects(new Error('error'));
+
+            return index.tagTemplate(config)
+                .then(() => assert.fail('should not get here'))
+                .catch(err => assert.equal(err.message, 'error'));
+        });
+
+        it('throws error for the corresponding request error status code if not 201', () => {
+            const responseFake = {
+                statusCode: 403,
+                body: {
+                    statusCode: 403,
+                    error: 'Forbidden',
+                    message: 'Fake forbidden message'
+                }
+            };
+
+            requestMock.resolves(responseFake);
+
+            return index.tagTemplate(config)
+                .then(() => assert.fail('should not get here'))
+                .catch(err => assert.equal(err.message,
```

Comment:
Could also combine the `catch` block with the `then` call.

```
   .then( assert.fail(...), (err) => assert.equal(...))
```

What would actually happen with the way it's currently written is that the `assert.fail` throws an error, which is then caught by the next `.catch()` block. So then it asserts that `'should not get here'` does not equal `'Error tagging....'`. It makes the error much more obvious.